File: algorithms/0104.Maximum Depth of Binary Tree/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
  public int maxDepth(TreeNode root) {
    return root == null ? 0 : 1 + Math.max(this.getDepth(root.left), this.getDepth(root.righ));
  }
}

File: algorithms/0104.Maximum Depth of Binary Tree/Solution.cs
using System;
using System.Collections.Generic;

namespace MaximumDepthOfBinaryTree
{
    public class Solution
    {
        public int MaxDepth(TreeNode root)
        {
            if (root == null)
            {
                return 0;
            }

            return Math.Max(MaxDepth(root.left), MaxDepth(root.right)) + 1;
        }

        public int MaxDepth1(TreeNode root)
        {
            var max = 0;
            var queue = new Queue<TreeNode>();
            queue.Enqueue(root);
            while (queue.Count != 0)
            {
                max++;
                var count = queue.Count;
                while (count-- != 0)
                {
                    var node = queue.Dequeue();
                    if (node == null)
                    {
                        continue;
                    }

                    queue.Enqueue(node.left);
                    queue.Enqueue(node.right);
                }
            }

            return --max;
        }
    }
}
